runtime/fp8_support.cu

```diff
@@ -149,6 +201,58 @@ __device__ __inline__ __e5m2 __float2e5m2(const float);
 struct __align__(1) __e5m2 {
   __e5m2() = default;
 
+  __e5m2(const __e5m2& other) {
```

Comment:
Off-topic: could `__half`, `__bfloat`, `__e4m3`, and `__e5m2` all be written as instantiations of a template:
```c++
template <typename Underlying>
struct __align__(sizeof(Underlying)) ReducedPrecisionType {
  // ...
 protected:
  Underlying __x;
};
using __half = ReducedPrecisionType<uint16_t>;
using __bfloat = ReducedPrecisionType<uint16_t>;
using __e4m3 = ReducedPrecisionType<uint8_t>;
using __e5m2 = ReducedPrecisionType<uint8_t>;
```